### ISSUE TITLE
Rewrite the LaTeX syntax

### ISSUE THREAD

**Issue by randy3k** [sublimehq/Packages#340]
The default LaTeX syntax is quite problematic, for examples
- #73, #300 
- https://github.com/jonschlinkert/sublime-markdown-extended/issues/119, 
- doesn't highlight `\newcommand{\bg}{\begin{equation}}` correctly.
- highlight `\text{}`and `\mbox{}` as string within math mode (changing color scheme does not fix this)
<img width="404" alt="screen shot 2016-04-30 at 3 22 02 pm" src="https://cloud.githubusercontent.com/assets/1690993/14938089/5636a752-0ee7-11e6-9013-18dd2dc12832.png">

There are 3 latex related packages now,
- https://github.com/SublimeText/LaTeXTools
- https://github.com/LaTeXing/LaTeXing
- https://github.com/randy3k/LaTeX-Plus

LaTeXTools uses the default LaTeX syntax and LaTeXing defines a wrapper syntax `LaTeXing` which imports the default LaTeX syntax. They both suffer both the same problems that the default LaTeX has.
Given the problems, I have created `LaTeX+` syntax from scratch in the package [LaTeX-Plus](https://github.com/randy3k/LaTeX-Plus/tree/master/syntax).

`LaTeX+` basically contains all features of the default `LaTeX` with the following two important changes.

1. It avoids the usage of $top_level_main, as indicted in #73, $top_level_main breaks a lot of things.

2. LaTeX math environments are highlighted as `string` scope, I found that it is a bad idea since it causes the problems of highlighting `\text{}`and `\mbox{}` as stated above.
In `LaTeX+`, the scope of math environment is defined as `meta.function.environment.math.latex` instead. Because of this change,  `LaTeX+` syntax is not compatible with `LaTeXTools` and `LaTeXing` commands which use scope `string.other.math.tex`. But the fix is easy, they just need to change the commands' scope from `string.other.math.tex` to `string.other.math.tex, meta.function.environment.math.latex`. This will ensure that the commands are compatible with `LaTeX+` and the default `LaTeX`.

To rewrite the LaTeX syntax, I suggest that we could use `LaTeX+` as a skeleton. @csch0 @msiniscalchi @ig0774 (the developers of LaTeXtools and LaTeXing) Do you have any comments?

**Comment by keith-hall** [comment]
I think it would be a good step in the right direction if you LaTeX package developers clubbed together to create some LaTeX syntax test files :)

**Comment by wbond** [comment]
I've got a good amount of experience in rewriting the syntaxes at this point that I can definitely work with some people here on this. The downside is that I know absolutely zero about TeX and LaTeX.

I believe these are the last two languages in the default packages that use the dreaded `$top_level_main` that breaks embedding. D may still use it, but does not require it.

Syntax test files that provide examples of all different variety of the different syntax is one of the most useful things.

In general, when writing the revised syntaxes, I've found using simpler regular expressions in `match` rules, combined with more complex `push`, `pop` and `set` rules tends to be able to better handle funky situations.
